### PR TITLE
use structured binding declarations

### PR DESCRIPTION
**Pull request by neheb**
Shorter.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

**Comment by MaxKellermann** [comment]
That's good stuff, but the way you wrote this, your code creates lots of temporary copies. You changed lots of references to values.
This is probably the reason why the MPD binary gets enlarged by 3 kB with this change.

**Comment by neheb** [comment]
That’s embarrassing. I’ll try to fix that.

**Comment by neheb** [comment]
Hmm I'm trying to figure out where the problem is. bloaty shows this:

```
4c4
<   13.6%  3.77Mi   0.0%       0    [section .debug_str]
---
>   13.6%  3.78Mi   0.0%       0    [section .debug_str]
12c12
<    0.3%  77.6Ki   0.8%  27.0Ki    std::get<>()
---
>    0.3%  78.0Ki   0.8%  27.1Ki    std::get<>()
123d122
<    0.0%  5.66Ki   0.0%       0    [Unmapped]
127a127
>    0.0%  5.38Ki   0.0%       0    [Unmapped]
143a144
>    0.0%  4.36Ki   0.0%  1.25Ki    std::__pair_get<>::__const_get<>()
157d157
<    0.0%  4.02Ki   0.0%  1.12Ki    std::__pair_get<>::__const_get<>()
1041a1042
>    0.0%     795   0.0%     720    ScanOpusTags()
1064d1064
<    0.0%     779   0.0%     704    ScanOpusTags()
4290d4289
<    0.0%     248   0.0%     134    Database
4318a4318
>    0.0%     247   0.0%     133    Database
8330a8331
>    0.0%      59   0.0%      23    GIT_VERSION
8342d8342
<    0.0%      58   0.0%      22    GIT_VERSION

```

**Comment by neheb** [comment]
I couldn't figure it out. Removed the ones I found problematic.

File: src/db/update/Walk.cxx [hunk 1]
```diff
@@ -415,21 +415,18 @@ UpdateWalk::DirectoryMakeUriParentChecked(Directory &root,
...
+		auto [name, rest] = uri.Split('/');
 		if (rest == nullptr)
 			break;
 
 		if (!name.empty()) {
-			directory = DirectoryMakeChildChecked(*directory,
-							      std::string(name).c_str(),
-							      s.first);
+			directory = DirectoryMakeChildChecked(
+				*directory, std::string(name).c_str(), name);
```
Comment (MaxKellermann) [inline src/db/update/Walk.cxx:424]: No change here, just reformatted the code (in a way that violates MPD's coding style).

File: src/lib/curl/Form.cxx [hunk 2]
```diff
@@ -36,16 +36,16 @@ EncodeForm(CURL *curl,
...
-		result.append(i.first);
+		result.append(key);
 		result.push_back('=');
 
-		if (!i.second.empty()) {
-			CurlString value(curl_easy_escape(curl, i.second.data(),
-							  i.second.length()));
+		if (!field.empty()) {
+			CurlString value(
+				curl_easy_escape(curl, field.data(), field.length()));
```
Comment (MaxKellermann) [inline src/lib/curl/Form.cxx:48]: Bad coding style

File: src/db/update/Walk.cxx (hunk 1 above)
Comment (neheb) [reply src/db/update/Walk.cxx:424]: I just ran got clang-tidy

File: src/lib/curl/Form.cxx (hunk 2 above)
Comment (neheb) [reply src/lib/curl/Form.cxx:48]: Same as above.

File: src/db/update/Walk.cxx (hunk 1 above)
Comment (MaxKellermann) [reply src/db/update/Walk.cxx:424]: Please stop hiding behind the tools you've chosen to use. It's not the tools' fault that you submit broken code often. Tools are just that - tools. You should manually review everything before you submit it. Everybody can use such a tool, everybody can run clang-tidy. The difficult part is understanding what happens and verify if what happens is good.

Comment (neheb) [reply src/db/update/Walk.cxx:424]: The goal is that all my PRs go through clang-tidy so as to have a consistent codebase.

I found something strange while playing around with .clang-format.

```
--- a/src/db/update/Walk.cxx
+++ b/src/db/update/Walk.cxx
@@ -432,9 +432,8 @@ UpdateWalk::DirectoryMakeUriParentChecked(Directory &root,
                        break;
 
                if (!name.empty()) {
-                       directory = DirectoryMakeChildChecked(*directory,
-                                                             std::string(name).c_str(),
-                                                             name);
+                       directory = DirectoryMakeChildChecked(
+                               *directory, std::string(name).c_str(), name);
                        if (directory == nullptr)
                                break;
                }
--- a/src/output/plugins/WasapiOutputPlugin.cxx
+++ b/src/output/plugins/WasapiOutputPlugin.cxx
@@ -322,7 +322,9 @@ void WasapiOutput::Enable() {
 
        if (enumerate_devices && SafeTry([this]() { EnumerateDevices(); })) {
                for (const auto &[device, desc] : device_desc) {
-                       FormatNotice(wasapi_output_domain, "Device \"%u\" \"%s\"", device,
+                       FormatNotice(wasapi_output_domain,
+                                    "Device \"%u\" \"%s\"",
+                                    device,
                                     desc.c_str());
                }
        }

```

That second one was caused by setting BinPackParameters to false.

As far as I can tell. clang-tidy works properly with calling functions directly but not when assigning them.

**Pull request merged**